gobblin-data-management/src/main/java/gobblin/data/management/copy/CopyableDatasetBase.java

```diff
@@ -0,0 +1,11 @@
+package gobblin.data.management.copy;
+
+import gobblin.dataset.Dataset;
+
+
+/**
+ * A common superinterface for {@link Dataset}s that can be operated on by distcp.
+ * Concrete classes must implement a subinterface of this interface ({@link CopyableDataset} or {@link IterableCopyableDataset}).
+ */
+interface CopyableDatasetBase extends Dataset {
```

Comment:
Just to have a superinterface for both `CopyableDataset` and `IterableCopyableDataset`. The alternative is to make `IterableCopyableDataset` to extend `CopyableDataset`, forcing users to implement the method `getCopyableFiles` which will never be used. I don't mind either way.